src/main/java/org/threadly/litesockets/tcp/ssl/SSLClient.java

```diff
@@ -246,37 +254,32 @@ public boolean isEncrypted() {
       if(ssle.getHandshakeStatus() == NEED_WRAP) {
         writeForce(ByteBuffer.allocate(0));
       }
-      if(this.ce != null) {
-        ce.getThreadScheduler().schedule(new Runnable() {
+      if(this.seb != null) {
```

Comment:
It was with how I was doing hasTimedout check on the tcpClient.

So the connection can fail for many reasons (connectionRefused, badAddress, badPort, Timedout).  And I need to be able to tell it which reason.  The once case watchdog would cover is the timeOut exception, and I want that to be very visible to anyone that gets it.